frontend/src/components/common/form/ParentSelect.tsx

```diff
@@ -0,0 +1,180 @@
+import { getIn, useFormikContext } from 'formik';
+import { groupBy, sortBy } from 'lodash';
+import React, { Fragment, useEffect, useState } from 'react';
+import { Highlighter, Menu, MenuItem } from 'react-bootstrap-typeahead';
+import { Label } from '../Label';
+import { SelectOption, SelectOptions } from './Select';
+import { TypeaheadField } from './Typeahead';
+
+interface IParentSelect {
+  /** specify the field that is being accessed */
+  field: string;
+  /** give the component the desired options to group by parent with */
+  options: SelectOptions;
+  /** if you would like placeholder text within the input */
+  placeholder?: string;
+  /** filterBy is optional - if used can determine what you can filter with */
+  filterBy?: string[];
+  /** provide the opitional label text*/
+  label?: string;
+  /** flag used to control whehter to display red astrix indicating required */
+  required?: boolean;
+  /** determine whether this component is disabled or not */
+  disabled?: boolean;
+  /** if this component is being used as a filter do not pre populate the value */
+  isFilter?: boolean;
+  /** set this flag if the grouped parent select will be used to make multiple selection */
+  enableMultiple?: boolean;
+  /** check to see whether reset button has been called to clear multiple selected */
+  clearSelected?: boolean;
+  /** reset state of clear selected via this component */
+  setClearSelected?: Function;
+}
+
+/** Component used to group children items with their parent.
+ *  Can click the parent item as a value or the subsequent children
+ **/
+export const ParentSelect: React.FC<IParentSelect> = ({
+  field,
+  options,
+  placeholder,
+  filterBy,
+  required,
+  disabled,
+  enableMultiple,
+  isFilter,
+  clearSelected,
```

Comment:
Is `clearSelected` a switch that each time it changes back and forth it results in refreshing the dropdown and removing the selected items?

Reply:
It is to trigger the` ref.current.clear` for the input to clear the multiple selections